tests/wrapper/test_bb_base.py

```diff
@@ -104,3 +105,56 @@ def test_multiple_wrappers(self):
         self.assertEqual(wrapper2._access_token, None)
 
         self.mocked_auth_requests.post.assert_called_once()
+
+    def test_authentication_fail_and_reauthentication(self):
+        """
+        Teste para verificar uma nova tentativa de autenticação
+        após uma falha na autenticação.
+
+        Dado:
+            -
+        Quando:
+            - for chamado BaseBBWrapper()._BaseBBWrapper__authenticate()
+        Então:
+            - 2 tentativas de autenticação devem ser realizadas,
+              uma falha e uma bem sucedida
+            - o resultado da autenticação deve ser True
+        """
+        bb_wrapper = BaseBBWrapper()
+
+        fail_attempts = 1
+        total_attempts = fail_attempts + 1
+
+        self.set_fail_auth(fail_attempts)
+
+        result = bb_wrapper._BaseBBWrapper__authenticate()
+
+        self.assertTrue(result)
+        self.assertEqual(total_attempts, self.mocked_auth_requests.post.call_count)
+
+    def test_authentication_fail_and_reauthentication_fail_after_5_attempts(self):
+        """
+        Teste para verificar falhas em todas as tentativas de autenticação.
+
+        Dado:
+            -
+        Quando:
+            - for chamado BaseBBWrapper()._BaseBBWrapper__authenticate()
+        Então:
+            - o máximo de tentativas de autenticação devem ser realizadas
+            - o resultado da autenticação deve ser False
+        """
+        bb_wrapper = BaseBBWrapper()
+
+        fail_attempts = bb_wrapper.MAX_ATTEMPTS
+        total_attempts = fail_attempts
+
+        self.set_fail_auth(fail_attempts)
+
+        try:
+            bb_wrapper._BaseBBWrapper__authenticate()
+        except HTTPError as err:
+            response = err.response
+            self.assertEqual(response.status_code, 401, msg=response.data)
```

Comment:
existe o assertRaises para lidar com testes que uma exceção deve ser levantada!

```suggestion
        with self.assertRaises(HTTPError) as ctx:
            bb_wrapper._BaseBBWrapper__authenticate()
        response = ctx.exception.response
        self.assertEqual(response.status_code, 401, msg=response.data)
            
```